### ISSUE TITLE
Make 0.x compatible with Python 3.13

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#2434]
Python 3.13 came out today: https://docs.python.org/3.13/whatsnew/3.13.html

Datasette does not run under Python 3.13 due to this issue with Pint:

- https://github.com/hgrecco/pint/issues/1969

Options:
- Ship a Datasette 0.x version that marks itself as not working with Python 3.13
- Fork Pint, fix the bug, ship a pint-for-python313 packago to PyPI and ship a Datasette 0.x version that depends on that
- Backport the removal of Pint from 1.0 to 0.x and ship a Datasette 0.x version that no longer has the units feature
- Vendor a fixed version of Pint into the 0.x branch

I like that last option best.

**Comment by simonw** [comment]
![CleanShot 2024-10-07 at 10 04 01@2x](https://github.com/user-attachments/assets/04ce7529-4e30-4b80-a06d-314d57277c8c)

Looks like Python 3.13 release isn't in there yet - see https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json which lists `3.13.0-rc.3`.

**Comment by simonw** [comment]
Got this failure on 3.8:

```
ERROR: Ignored the following versions that require a different python version: 0.3 Requires-Python >=3.9
ERROR: Could not find a version that satisfies the requirement flexcache>=0.3 (from datasette) (from versions: 0.1, 0.2)
ERROR: No matching distribution found for flexcache>=0.3
```
I'm going to drop 3.8 from the support matrix and see if I can get the tests to pass.

**Comment by simonw** [comment]
![CleanShot 2024-10-07 at 10 09 13@2x](https://github.com/user-attachments/assets/db03ec84-788a-4d83-abcc-5cbb9d1b2d28)

Don't spell check in `vendored`.

**Comment by simonw** [comment]
Also got this error in the Pyodide tests:

![CleanShot 2024-10-07 at 10 10 47@2x](https://github.com/user-attachments/assets/5c7355fc-44d6-4c45-ba3d-84d6f68b3391)

I think there are files in `vendored` that are not being included in the wheel.

**Comment by simonw** [comment]
Looks like Pint itself dropped support for 3.8 here: https://github.com/hgrecco/pint/commit/1208b3d503913eabe8ac96db18ee2ad14af1bdd6

Which I think was released in 0.22 in May 2023 https://github.com/hgrecco/pint/releases/tag/0.22

**Comment by simonw** [comment]
I'm OK with shipping a version that drops support for 3.8 while adding support for 3.13.

**Comment by simonw** [comment]
Released: https://docs.datasette.io/en/0.65/changelog.html#v0-65

**Comment by simonw** [comment]
Tested it like this:

```
$ mkdir python-3.13 
$ cd python-3.13 
$ uv venv --python 3.13
Using CPython 3.13.0rc2
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
$ source .venv/bin/activate
$ uv run which python
/private/tmp/python-3.13/.venv/bin/python
$ uv pip install datasette
Resolved 29 packages in 1.36s
Prepared 1 package in 190ms
...
$ uv run which datasette
/private/tmp/python-3.13/.venv/bin/datasette
$ uv run datasette --get /-/versions.json | jq
{
  "python": {
    "version": "3.13.0",
    "full": "3.13.0rc2 (main, Sep  9 2024, 22:09:15) [Clang 18.1.8 ]"
  },
  "datasette": {
    "version": "0.65"
  },
  "asgi": "3.0",
  "uvicorn": "0.31.0",
  "sqlite": {
    "version": "3.46.0",
```
